iOS/SampleApp/SampleApp/Services/ManagingJSON.swift

```diff
@@ -0,0 +1,84 @@
+// Copyright 2019 Flying Vehicle Monster team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+extension ViewController {
+    internal func prepareDamagedCarPartsToSend() -> String {
+        func removeRedundantCharacters() -> String {
+            return CurrentValue.DamagedCarParts.replacingOccurrences(of: " ", with: ";").replacingOccurrences(of: ",", with: ";")
+        }
+        
+        func parseDamagedCarParts(toParse tmp: String) -> String {
+            let elems = tmp.split(separator: ";")
+            var result = ""
+            var first = true
+            for e in elems {
+                if first {
+                    first = false
+                }
+                else {
+                    result.append(",")
+                }
+                
+                result.append("{ \"id\" : \"")
+                result.append(String(e))
+                result.append("\" }")
+            }
+            return result
+        }
+        
+        return parseDamagedCarParts(toParse: removeRedundantCharacters())
+    }
+    
+    internal func fillDamagedCarPartsTextView (_ returningJSON: String) {
+        func removeRedundantCharacters(_ tmp: String) -> String {
+            return tmp.replacingOccurrences(of: "\"", with: "").replacingOccurrences(of: ",", with: "").replacingOccurrences(of: "{", with: "").replacingOccurrences(of: "}", with: "").replacingOccurrences(of: "]", with: "").replacingOccurrences(of: "[", with: "").replacingOccurrences(of: "id", with: "")
+        }
+        
+        let tmp = returningJSON.split(separator: ":")
+        var i = 0
+        var result = ""
```

Comment:
What the hell is that?!

Reply:
I mean whole this file

Reply:
Could you explain me what problems are in this file?

Reply:
do we really need it? It smells a lot

Reply:
Yes, it ensures proper sending and receiving "data" between app